src/scripts/manage_artifacts/artifacts.py

```diff
@@ -1,78 +1,86 @@
-  Uploading to S3 and deletion from S3 of scripts and templates.
-  '''
   parser = argparse.ArgumentParser(
-    description='')
-  parser.add_argument(
-    '--ref')
+    description='Uploading to S3 and deletion from S3 of scripts and templates.',
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+  parser.add_argument('--ref')
   group = parser.add_mutually_exclusive_group(required=True)
   group.add_argument('--upload', action='store_true')
   group.add_argument('--remove', action='store_true')
   group.add_argument('--list', action='store_true')
+  parser.add_argument(
+          '--environment',
+          default="develop",
+          choices=["develop", "prod"],
+          help="The deployment environment")
   args = parser.parse_args()
   return args
 
 
 def execute_command(cmd):
   print(f'Invoking command: {" ".join(cmd)}')
-  subprocess.run(cmd)
+  subprocess.run(cmd, check=True)
 
 
-def upload(ref):
+def upload(ref, cfn_bucket):
   '''Copy Glue scripts to the artifacts bucket'''
   scripts_local_path = 'src/glue/'
-  scripts_s3_path = f's3://{cfn_bucket}/{repo_name}/{ref}/glue/'
+  scripts_s3_path = f's3://{cfn_bucket}/{REPO_NAME}/{ref}/glue/'
   cmd = ['aws', 's3', 'sync', scripts_local_path, scripts_s3_path]
   execute_command(cmd)
 
   '''Copies Lambda code and template to the artifacts bucket'''
   lambda_local_path = 'src/lambda/'
-  lambda_s3_path = f's3://{cfn_bucket}/{repo_name}/{ref}/lambda/'
+  lambda_s3_path = f's3://{cfn_bucket}/{REPO_NAME}/{ref}/lambda/'
   cmd = ['aws', 's3', 'sync', lambda_local_path, lambda_s3_path]
   execute_command(cmd)
 
   '''Copy EC2 resources (e.g., crontab) to the artifacts bucket'''
   resources_local_path = 'src/ec2/resources/'
-  resources_s3_path = f's3://{cfn_bucket}/{repo_name}/{ref}/ec2/resources/'
+  resources_s3_path = f's3://{cfn_bucket}/{REPO_NAME}/{ref}/ec2/resources/'
   cmd = ['aws', 's3', 'sync', resources_local_path, resources_s3_path]
   execute_command(cmd)
 
   '''Copy CFN templates to the artifacts bucket'''
   templates_local_path = 'templates/'
-  templates_s3_path = f's3://{cfn_bucket}/{repo_name}/{ref}/templates/'
+  templates_s3_path = f's3://{cfn_bucket}/{REPO_NAME}/{ref}/templates/'
   cmd = ['aws', 's3', 'sync', templates_local_path, templates_s3_path]
   execute_command(cmd)
 
-def delete(ref):
+def delete(ref, cfn_bucket):
   '''Removes all files recursively for ref'''
-  s3_path = f's3://{cfn_bucket}/{repo_name}/{ref}/'
+  s3_path = f's3://{cfn_bucket}/{REPO_NAME}/{ref}/'
   cmd = ['aws', 's3', 'rm', '--recursive', s3_path]
   execute_command(cmd)
 
 
-def list_refs():
+def list_refs(cfn_bucket):
   '''List all refs'''
-  s3_path = f's3://{cfn_bucket}/{repo_name}/'
+  s3_path = f's3://{cfn_bucket}/{REPO_NAME}/'
   cmd = ['aws','s3','ls', s3_path]
   execute_command(cmd)
 
 
-def main(args):
+def main():
+  args = read_args()
+  if args.environment == "prod":
+    cfn_bucket = "sceptre-cloudformation-bucket-bucket-10mwvvuhlvtk9"
```

Comment:
Out of curiosity, was there a significance to the prod/develop cfn buckets named this way?

Reply:
No, these buckets were created by IT along with the account. They were already there so we started using them for some of our artifact stuff.